### PR TITLE
Do not add secret finalizer while deleting a host

### PR DESCRIPTION
**Pull request by andfasano**
Fixes https://github.com/metal3-io/baremetal-operator/issues/1053

### CODE REVIEW

**Comment by s3rj1k** [comment]
@andfasano nice catch, thanks!

**Comment by andfasano** [comment]
> @andfasano nice catch, thanks!

Thanks. Unfortunately it happens intermittently, so it took some time to nail it down. I also tested it functionally (manually) and it seems ok.

File: controllers/metal3.io/baremetalhost_controller_test.go [hunk 1]
```diff
@@ -455,6 +455,48 @@ func TestAddFinalizers(t *testing.T) {
 	)
 }
 
+// TestDoNotAddSecretFinalizersDuringDelete verifies that during an host deletion,
```
Comment (bfournie) [inline controllers/metal3.io/baremetalhost_controller_test.go:458]: Nit - 'an' should be 'a'

File: controllers/metal3.io/baremetalhost_controller_test.go [hunk 2]
```diff
@@ -455,6 +455,48 @@ func TestAddFinalizers(t *testing.T) {
...
+// in case the removal of a secret finalizer triggers an immediate reconcile loop,
+// then the secret finalizer is not again
```
Comment (bfournie) [inline controllers/metal3.io/baremetalhost_controller_test.go:460]: Nit - seems like some text is missing here ("finalizer is not again")

**Review by bfournie — APPROVED** [review]

**Review by elfosardo — APPROVED** [review]

**Pull request merged**